src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -1677,11 +1667,18 @@ def _get_hyper_param_config(self):
         if self.training_params.log_installed_packages:
             pkg_list = list(map(lambda pkg: str(pkg), _get_installed_distributions()))
             additional_log_items["installed_packages"] = pkg_list
+
+        dataset_params = {
+            "train_dataset_params": self.train_loader.dataset.dataset_params if hasattr(self.train_loader.dataset, "dataset_params") else None,
+            "train_dataloader_params": self.train_loader.dataloader_params if hasattr(self.train_loader, "dataloader_params") else None,
+            "valid_dataset_params": self.valid_loader.dataset.dataset_params if hasattr(self.valid_loader.dataset, "dataset_params") else None,
+            "valid_dataloader_params": self.valid_loader.dataloader_params if hasattr(self.valid_loader, "dataloader_params") else None,
+        }
         hyper_param_config = {
             "arch_params": self.arch_params.__dict__,
             "checkpoint_params": self.checkpoint_params.__dict__,
             "training_hyperparams": self.training_params.__dict__,
-            "dataset_params": self.dataset_params.__dict__,
+            "dataset_params": dataset_params,
```

Comment:
We are storing here non-instantiated config, do we?

Reply:
Yeah it's the non-instantiated config:
```
{'train_dataset_params': {'root': './data/cifar10', 'train': True, 'transforms': [{'RandomCrop': {'size': 32, 'padding': 4}}, 'RandomHorizontalFlip', 'ToTensor', {'Normalize': {'mean': [0.4914, 0.4822, 0.4465], 'std': [0.2023, 0.1994, 0.201]}}], 'target_transform': None, 'download': True},
 'train_dataloader_params': {'shuffle': True, 'batch_size': 256, 'num_workers': 8, 'drop_last': False, 'pin_memory': True},
 'valid_dataset_params': {'root': './data/cifar10', 'train': False, 'transforms': ['ToTensor', {'Normalize': {'mean': [0.4914, 0.4822, 0.4465], 'std': [0.2023, 0.1994, 0.201]}}], 'target_transform': None, 'download': True},
 'valid_dataloader_params': {'batch_size': 512, 'num_workers': 8, 'drop_last': False, 'pin_memory': True}}
 ```